### DISCUSSION TITLE
Appveyor integration documnetation

### DISCUSSION THREAD

**Discussion by HarelM** [fastlane/fastlane#16855] [Ideas]
It would be great if appveyor (which now also has macOS environment) will be added to the CI list in the documentation.
<img width="246" alt="Screen Shot 2020-07-14 at 21 38 26" src="https://user-images.githubusercontent.com/3269297/87463831-c1238000-c61a-11ea-9c08-f880c434083f.png">

**Comment by joshdholtz** [comment]
@HarelM That's cool! I didn't know they added an macOS environment. Do you happen to hav any experience setting _fastlane_ up to run in Appveyor's macOS environment?

**Comment by HarelM** [comment]
I'm struggling with it at the moment...
I'm running appveyor for a long time now and I'm very experienced with it, but I have zero experience with fastlane...

**Reply by joshdholtz** [reply]
I’d be happy to work through some issue you are running into here if you don’t mind having that conversation 😊 I have quiet a few other things I’m juggling at the moment so I’m not able to put appveyor on my list at the moment but I’d be happy to work with you on some issues here.

**Reply by HarelM** [reply]
I'd be happy to receive any help.
The issue I'm tracking for this is this one:
https://github.com/IsraelHikingMap/Site/issues/1266
At first stage, just to get things running I would like to upload the ipa file I'm already creating to testflight.
I'm using the following branch:
https://github.com/IsraelHikingMap/Site/tree/fastlane-1266
Most of the related files can be seen here:
https://github.com/IsraelHikingMap/Site/tree/fastlane-1266/IsraelHiking.Web/fastlane
Build outputs can be found here:
https://ci.appveyor.com/project/IsraelHikingHost/site

My current problem is that I'm struggling with the Ruby installation on the CI mac.
I've read that it's not good to use the system Ruby and I should install it, probably using rbenv, but I'm getting tons of errors and warnings so I'm not sure what's working and what's not working properly.
See here:
https://ci.appveyor.com/project/IsraelHikingHost/site?fullLog=true
```
installing bundler
Fetching bundler-2.1.4.gem
ERROR:  While executing gem ... (Errno::EACCES)
    Permission denied @ dir_s_mkdir - /Users/appveyor/.rvm/gems/ruby-2.6.5/gems/bundler-2.1.4
```
I'm basically doing the following commands:
https://github.com/IsraelHikingMap/Site/blob/667eec3b77b70a252108ba279eb0f10da21b0e9d/Scripts/BuildAndPublishIOS.ps1#L4L15

**Reply by HarelM** [reply]
Down the rabbit hole...
I have changed the relevant fastlane and ruby commands to use sudo, otherwise it's a hassle to make it work...
I've added the `FASTLANE_USER, FASTLANE_PASSWORD, FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD` env variable to the CI code. With `sudo` and these variables the CI seems to be stuck when asking for the password for the app connect account, probably due to needed user interaction.
This account has two factor authentication enabled.
@joshdholtz Let me know if you need more info. I'm not sure how to proceed here...

**Reply by HarelM** [reply]
Let me know if there's another channel I can communicate more frequently as I'm pretty stuck right now... :-(

**Reply by HarelM** [reply]
Still doesn't work.
I've tried to add `setup_ci` but it now fails on a different issue.
Latest run:
https://ci.appveyor.com/project/IsraelHikingHost/site/builds/34171010
Latest code can be seen here:
https://github.com/IsraelHikingMap/Site/tree/fastlane-1266/IsraelHiking.Web/fastlane

**Reply by joshdholtz** [reply]
@HarelM Sorry for slow response! I had to go heads down on some feature development end of last week but now I’m back to monitoring my GitHub notifications faster!

> This account has two factor authentication enabled.

Ahhh yeah, this is a touch issue with CI 😔 Solutions for this can be found (look for `FASTLANE_SESSION `):
- http://docs.fastlane.tools/best-practices/continuous-integration/#two-step-or-two-factor-auth
- http://docs.fastlane.tools/best-practices/continuous-integration/#environment-variables-to-set

> Latest run:

I took a look at this output and it looks like there is an issue on Appveyor with setting up keychain. The command is failing for some reason 🤔 `Shell command exited with exit status 1 instead of 0` should not be happening.  I’ve sent them a tweet to see if I can get some more info from them 👉  https://twitter.com/joshdholtz/status/1285253948799156225?s=21

**Reply by HarelM** [reply]
Thanks for getting back to me!
The two things I would like to achieve with fastlane is matching and upload so I don't think I need to use fastlane session, or at least I hope not to need to use it...
I'm able to manipulate the keychain as can be seen in my code here which run in my CI:
https://github.com/IsraelHikingMap/Site/blob/54e246219345246f2445171080195148bcfba3c0/Scripts/BuildAndPublishIOS.ps1#L19L48
So I'm not 100% sure it's an appveyor issue. And also I'm getting a similar issue when doing the same on my mac (same error code, but a bit different keychain (login instead of tmp).
Let me know what I can test and I'll be happy to run it.

**Comment by HarelM** [comment]
@joshdholtz any updates on this? I'm really stuck... :-(
